.github/workflows/release.yml

```diff
@@ -0,0 +1,79 @@
+name: release
+on:
+  push:
+    tags:
+      - 'v*'
+    branches:
+      - main
+      - dev-*
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-binaries:
+    runs-on: ubuntu-latest
+    if: github.repository == 'numaproj-labs/numaplane'
+    name: Build binaries
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Go
```

Comment:
actually, do we need Go? The binary is being compiled within the Docker context on top of [golang](https://github.com/numaproj-labs/numaplane/blob/main/Dockerfile#L2), right?